packages/itwin/property-grid/src/hooks/UseDataProvider.ts

```diff
@@ -0,0 +1,65 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { useCallback, useEffect } from "react";
+import { useDisposable } from "@itwin/core-react";
+import { KeySet } from "@itwin/presentation-common";
+import { PresentationPropertyDataProvider, usePropertyDataProviderWithUnifiedSelection } from "@itwin/presentation-components";
+
+import type { IModelConnection } from "@itwin/core-frontend";
+import type { InstanceKey } from "@itwin/presentation-common";
+
+/** Props for configuring data providers used by `PropertyGrid`. */
+export interface DataProviderProps {
+  /** Flag that specified whether properties favorite properties support should be enabled or not. */
+  enableFavoriteProperties?: boolean;
+  /** Flag that specified whether nested property categories are enabled. */
+  enablePropertyGroupNesting?: boolean;
+  /** Id of ruleset that should be used when pulling data. */
+  rulesetId?: string;
+  /** Callback that creates custom data provider that should be used instead of default one. */
+  createDataProvider?: (imodel: IModelConnection) => PresentationPropertyDataProvider;
+}
+
```

Comment:
I believe this should be a union:
```ts
{
  /** Flag that specified whether properties favorite properties support should be enabled or not. */
  enableFavoriteProperties?: boolean;
  /** Flag that specified whether nested property categories are enabled. */
  enablePropertyGroupNesting?: boolean;
  /** Id of ruleset that should be used when pulling data. */
  rulesetId?: string;
} | {
  /** Callback that creates custom data provider that should be used instead of default one. */
  createDataProvider: (imodel: IModelConnection) => PresentationPropertyDataProvider;
}
```

Also, I guess `createDataProvider` could return `IPresentationPropertyDataProvider`?